### ISSUE TITLE
e2e BeforeSuite flaky in GKE

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#21322]
Looks like pods aren't getting scheduled:

```
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod heapster-v14-s9yp3 is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod kube-dns-v10-ae96i is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod kube-dns-v10-d778v is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod kubernetes-dashboard-v0.1.0-1dhse is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod kubernetes-dashboard-v0.1.0-vwg29 is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod l7-lb-controller-m2zcg is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: The status of Pod l7-lb-controller-orqra is Pending, waiting for it to be either Running or Failed
08:06:05 Feb 16 08:06:05.081: INFO: 5 / 12 pods in namespace 'kube-system' are running and ready (442 seconds elapsed)
08:06:05 Feb 16 08:06:05.081: INFO: expected 4 pod replicas in namespace 'kube-system', 1 are Running and Ready.
08:06:05 Feb 16 08:06:05.081: INFO: POD                                NODE                                           PHASE    GRACE  CONDITIONS
08:06:05 Feb 16 08:06:05.081: INFO: heapster-v14-s9yp3                 gke-jenkins-gke-e2e-serial-7da628fa-node-o9j1  Pending  30s    []
08:06:05 Feb 16 08:06:05.081: INFO: kube-dns-v10-ae96i                 gke-jenkins-gke-e2e-serial-7da628fa-node-uesa  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-02-16 08:05:57 -0800 PST ContainersNotReady containers with unready status: [etcd kube2sky skydns healthz]}]
08:06:05 Feb 16 08:06:05.081: INFO: kube-dns-v10-d778v                 gke-jenkins-gke-e2e-serial-7da628fa-node-o9j1  Pending  30s    []
08:06:05 Feb 16 08:06:05.081: INFO: kubernetes-dashboard-v0.1.0-1dhse  gke-jenkins-gke-e2e-serial-7da628fa-node-o9j1  Pending  30s    []
08:06:05 Feb 16 08:06:05.081: INFO: kubernetes-dashboard-v0.1.0-vwg29  gke-jenkins-gke-e2e-serial-7da628fa-node-78k4  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-02-16 08:05:58 -0800 PST ContainersNotReady containers with unready status: [kubernetes-dashboard]}]
08:06:05 Feb 16 08:06:05.081: INFO: l7-lb-controller-m2zcg             gke-jenkins-gke-e2e-serial-7da628fa-node-o9j1  Pending  600s   []
08:06:05 Feb 16 08:06:05.082: INFO: l7-lb-controller-orqra             gke-jenkins-gke-e2e-serial-7da628fa-node-78k4  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-02-16 08:05:58 -0800 PST ContainersNotReady containers with unready status: [default-http-backend l7-lb-controller]}]
```

See e.g.

- http://test-ci.k8s.io/kubernetes-e2e-gke-serial/537
- http://test-ci.k8s.io/kubernetes-e2e-gke-serial/527
- http://test-ci.k8s.io/kubernetes-e2e-gke-serial/503

**Comment by a-robinson** [comment]
This has been flaking quite often today, P0 is very appropriate.

**Comment by roberthbailey** [comment]
Is this *only* flaky in GKE or is it also flaky in GCE?

**Comment by gmarek** [comment]
It's only flaky in GKE. It happens roughly in half of the runs and is main reason for merge queue blockages. I'd made it P-1 if there was something like this...

If it persists for another day I'd consider removing GKE from blockers until it's fixed.

**Comment by a-robinson** [comment]
cc @jlowdermilk

**Comment by jlowdermilk** [comment]
FYI, gke-serial already does not block merge queue.

We should really be printing events for kube-system. The console logs from failed runs show kube-system pods stuck in pending, but there's no indication as to why.

**Comment by a-robinson** [comment]
Sorry for not making that more clear - while @ihmccreery only linked to gke-serial failures, it's been affecting kubernetes-e2e-gke heavily lately.

Examples from just the last 12 hours:
https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gke/1197/
https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gke/1195/
https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gke/1188/

**Comment by ihmccreery** [comment]
Is it possible this is related to #20916?

**Comment by jlowdermilk** [comment]
From kubelet logs, it looks like docker problems on the nodes
https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gke/1197/

All three kubelet.log files (from [/artifacts](https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gke/1197/artifacts)) are full of
```
E0217 10:19:36.559859    3996 kubelet.go:2617] Container runtime sanity check failed: docker: failed to get docker version: cannot connect to Docker endpoint
E0217 10:19:36.575164    3996 generic.go:164] GenericPLEG: Unable to retrieve pods: cannot connect to Docker endpoint
```

Also looks like a some of the system pods were eventually started, but got killed for failing healthz.

cc @kubernetes/goog-node

**Comment by alex-mohr** [comment]
I saw the same with #21401.  If so, we either need to add a script to startup to work around the docker issue, or build a new containervm with 1.9.1?

**Comment by pwittrock** [comment]
Just saw a flake that looks like it maybe a separate issue:  Should this be filed separately?

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gke/1220

>
11:01:36 ERROR: (gcloud.container.clusters.create) Operation [<Operation
11:01:36  name: u'operation-1455734333499-a29e3a49'
11:01:36  operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
11:01:36  selfLink: u'https://test-container.sandbox.googleapis.com/v1/projects/249458424157/zones/us-central1-f/operations/operation-1455734333499-a29e3a49'
11:01:36  status: StatusValueValuesEnum(DONE, 3)
11:01:36  statusMessage: u'All cluster resources were brought up, but the cluster API is reporting that node "gke-jkns-gke-e2e-ci-e3558aae-node-v9m4" is unhealthy.'
11:01:36  targetLink: u'https://test-container.sandbox.googleapis.com/v1/projects/249458424157/zones/us-central1-f/clusters/jkns-gke-e2e-ci'
11:01:36  zone: u'us-central1-f'>] finished with error: All cluster resources were brought up, but the cluster API is reporting that node "gke-jkns-gke-e2e-ci-e3558aae-node-v9m4" is unhealthy. 
11:01:36 2016/02/17 11:01:36 e2e.go:307: Error running up: exit status 1
11:01:36 2016/02/17 11:01:36 e2e.go:303: Step 'up' finished in 22m44.815917361s
11:01:36 
2016/02/17 11:01:36 e2e.go:141: Error starting e2e cluster. Aborting.
11:01:36 exit status 1

Full console text:
[e2e-gke-flake.txt](https://github.com/kubernetes/kubernetes/files/135016/e2e-gke-flake.txt)

**Comment by jlowdermilk** [comment]
@pwittrock that looks like a different failure mode I've been trying to track down/repo. It might be related to docker/node problems, but feel free to file separately.

**Comment by pwittrock** [comment]
@jlowdermilk filed #21411

**Comment by pwittrock** [comment]
Filed #21450 for a similar issue but different console error messages.

**Comment by jlowdermilk** [comment]
This looks to be fixed with the new cvm that has docker 1.9.1 (#21514)
